ns1/permissions.go

```diff
@@ -235,6 +285,38 @@ func permissionsToResourceData(d *schema.ResourceData, permissions account.Permi
 
 func resourceDataToPermissions(d *schema.ResourceData) account.PermissionsMap {
 	var p account.PermissionsMap
+	if v, ok := d.GetOk("dns_records_allow"); ok {
+		records := []account.Record{}
```

Comment:
We already know what the capacity of this slice should be based on `len(schemaRecord)`.  We should allocate the slice with capacity defined for performance.

Reply:
I also tried to use make([]account.PermissionsRecord, len(schemaRecord)) but is not working properly. I think in this situation considering that the provider will not have a big charge it's fine to use this way, what do you think?